### PR TITLE
Use StackSummary where possible

### PR DESCRIPTION
**Pull request by Sterbic**
Solves the following use case:

```
import sys
import traceback
exc_info = sys.exc_info()
last_frame = traceback.extract_tb(exc_info[2])[-1]
print(last_frame.lineno)
mypy: test.py:5: error: "Tuple[str, int, str, Optional[str]]" has no attribute "lineno"
```

### CODE REVIEW

File: stdlib/2and3/traceback.pyi [hunk 1]
```diff
@@ -82,7 +88,7 @@ if sys.version_info >= (3, 5):
 
 
 if sys.version_info >= (3, 5):
-    class StackSummary:
+    class StackSummary(List):
```
Comment (JelleZijlstra) [inline stdlib/2and3/traceback.pyi]: Should this inherit from `List[_PT]`?

Comment (Sterbic) [inline stdlib/2and3/traceback.pyi]: Wouldn't that defeat the example I put in the description? _PT is a tuple while the elements of the list are frame objects.

Comment (JelleZijlstra) [inline stdlib/2and3/traceback.pyi]: Oh, it's actually a list of FrameSummary objects, so you should inherit from `List[FrameSummary]`.

It seems like the CPython documentation is wrong here; I filed http://bugs.python.org/issue30984 to fix it.

Comment (Sterbic) [inline stdlib/2and3/traceback.pyi]: Ok, will send an update!

**Pull request merged**